var/spack/repos/builtin/packages/edm4hep/package.py

```diff
@@ -68,7 +68,7 @@ class Edm4hep(CMakePackage):
     depends_on("nlohmann-json@3.10.5:")
     depends_on("podio@1:", when="@0.99:")
     depends_on("podio@0.15:", when="@:0.10.5")
-    for _std in _cxxstd_values:
+    for _std in ("17", "20"):
```

Comment:
I don't really like the repetition, but I guess we can live with it :-)

I think, you could define `_cxxstd_values = (conditional("17", when="@:0.99"), conditional("20"))` to have them both be same type (I'm assuming the conditional without a condition will work). Then you can loop
```suggestion
    for _cxxstd in _cxxstd_values:
        for _v in _cxxstd:
            depends_on(f"podio cxxstd={_v.value}", when=f"cxxstd={_v.value}")
```

Reply:
I wasn't aware that was even possible, so I didn't consider it. Done as suggested here now. **More importantly I also caught a version issue: it should be `conditional("17", when=@0.99.0)` to really work as expected**.